### PR TITLE
Simplify UDP test, use dynamic ports

### PR DESCRIPTION
**Pull request by josephnoir**
This PR simplifies a UDP test that regularly failed on our Jenkins Windows VMs.

### CODE REVIEW

File: libcaf_io/test/remote_actor_udp.cpp [hunk 1]
```diff
@@ -207,7 +207,12 @@ CAF_TEST(multiple_endpoints_udp) {
       }
     };
   });
-  server_sys.middleman().publish_udp(mirror, 12345);
+  // Publish server actor.
+  auto eport = server_sys.middleman().publish_udp(mirror, 0);
```
Comment (Neverlord) [inline libcaf_io/test/remote_actor_udp.cpp]: Simpler:

`auto port = unbox(server_sys.middleman().publish_udp(mirror, 0));`

File: libcaf_io/test/remote_actor_udp.cpp [hunk 2]
```diff
@@ -220,32 +225,25 @@ CAF_TEST(multiple_endpoints_udp) {
...
-  // setup client a
-  CAF_MESSAGE("creating first client");
-  config client_cfg;
-  actor_system client_sys{client_cfg};
-  auto client = client_sys.spawn(client_fun);
-  // acquire remote actor from the server
-  auto client_srv = client_sys.middleman().remote_actor_udp("localhost", 12345);
-  CAF_REQUIRE(client_srv);
-  // setup other clients
-  for (int i = 0; i < 5; ++i) {
```
Comment (Neverlord) [inline libcaf_io/test/remote_actor_udp.cpp]: Seems like the unit test now reduced coverage by using a single client instead of 5. What's the motivation for this change? More is better, when it comes to testing. 😉

Comment (josephnoir) [reply libcaf_io/test/remote_actor_udp.cpp]: It looks like the tests take too long I wanted to check if this makes a difference. Then I remembered that our PR builder doesn't build on Windows. I'm not sure what the good solution there is as I can't reproduce the error manually.

File: libcaf_io/test/remote_actor_udp.cpp (hunk 1 above)
Comment (josephnoir) [reply libcaf_io/test/remote_actor_udp.cpp]: It seems to be local to the [config_option test](https://github.com/actor-framework/actor-framework/blob/master/libcaf_core/test/config_option.cpp#L91).

File: libcaf_io/test/remote_actor_udp.cpp (hunk 2 above)
Comment (josephnoir) [reply libcaf_io/test/remote_actor_udp.cpp]: Just saw your comment in Gitter. I'll update the PR.

File: libcaf_io/test/remote_actor_udp.cpp (hunk 1 above)
Comment (Neverlord) [reply libcaf_io/test/remote_actor_udp.cpp]: Nope, just include `dsl.hpp`: https://github.com/actor-framework/actor-framework/blob/master/libcaf_test/caf/test/dsl.hpp#L627

Comment (Neverlord) [reply libcaf_io/test/remote_actor_udp.cpp]: Sorry, my bad. You're not using `test_coordinator_fixture`.
